clients/src/main/java/org/apache/kafka/common/security/oauthbearer/internals/OAuthBearerClientInitialResponse.java

```diff
@@ -34,7 +34,7 @@
     private static final String VALUE = "[\\x21-\\x7E \t\r\n]+";
 
     private static final String KVPAIRS = String.format("(%s=%s%s)*", KEY, VALUE, SEPARATOR);
-    private static final Pattern AUTH_PATTERN = Pattern.compile("(?<scheme>[\\w]+)[ ]+(?<token>[-_\\.a-zA-Z0-9]+)");
+    private static final Pattern AUTH_PATTERN = Pattern.compile("(?<scheme>[\\w]+)[ ]+(?<token>[+-_\\.a-zA-Z0-9]+[=]*)");
```

Comment:
This pattern is used to match bear token in base64 string. But the original AUTH_PATTERN is not working. According to base64 standard, a base64 string should also include characters like '+' and '=', the latter is used for  padding. In our dSTS scenarios, the base64 string of a dSTS token includes '=' and optionally a couple of '=' appended to the end.

Reply:
Please add this as documentation to future reference.